main.cpp

```diff
@@ -357,26 +397,26 @@ void InvoiceManager::Invoice::markAsPaid(Invoice& invoiceA)
     invoiceA.overdue = false;
 }
 
-void InvoiceManager::Invoice::duplicate(Invoice& invoiceA)
+void InvoiceManager::Invoice::duplicate(const Invoice& invoiceA)
 {
-    Invoice invoiceB = invoiceA;    // I would not really do this
-    ++invoiceB.invoiceNumber;
-    std::cout << "Duplicated invoice " << invoiceA.invoiceNumber << " as " << invoiceB.invoiceNumber << std::endl;
+    Invoice* invoiceB = new Invoice(invoiceA);
+    ++invoiceB->invoiceNumber;
+    std::cout << "Duplicated invoice " << invoiceA.invoiceNumber << " as " << invoiceB->invoiceNumber << std::endl;
 }
 
-InvoiceManager::Invoice InvoiceManager::createInvoice(std::string name, float date, std::string type = "post", float time = 0.0f)
+InvoiceManager::Invoice* InvoiceManager::createInvoice(const std::string name, const float date, const std::string type = "post", const float time = 0.0f)
 {
-    Invoice newInvoice(name);
+    Invoice* newInvoice = new Invoice(name);
     ++numInvoices;
-    newInvoice.invoiceNumber = numInvoices;
-    newInvoice.dueDate = date;
-    newInvoice.workType = type;
-    newInvoice.workTime = time;
+    newInvoice->invoiceNumber = numInvoices;
+    newInvoice->dueDate = date;
+    newInvoice->workType = type;
+    newInvoice->workTime = time;
 
     return newInvoice;
 }
```

Comment:
Don't allocate this object on the heap. 
You are not de-allocating it anywhere.
The point of this assignment is to learn how to manually manage heap-allocated objects in a safe manner using Wrapper classes. 
- [ ] revert to returning the invoice as a regular stack object.